diff -Naur node_modules/@angular/router/esm/src/segments.js ../@angular-RC/router/esm/src/segments.js
--- node_modules/@angular/router/esm/src/segments.js    2016-05-03 23:24:39.000000000 +0200
+++ ../@angular-RC/router/esm/src/segments.js   2016-05-16 17:02:14.560429830 +0200
@@ -130,6 +130,7 @@
         return false;
     if (a.outlet != b.outlet)
         return false;
+    if(!a.parameters && !b.parameters) return false;
     return StringMapWrapper.equals(a.parameters, b.parameters);
 }
 export function equalUrlSegments(a, b) {
diff -Naur node_modules/@angular/router/src/segments.js ../@angular-RC/router/src/segments.js
--- node_modules/@angular/router/src/segments.js    2016-05-03 23:24:33.000000000 +0200
+++ ../@angular-RC/router/src/segments.js   2016-05-16 17:02:20.520496172 +0200
@@ -171,6 +171,7 @@
         return false;
     if (a.outlet != b.outlet)
         return false;
+    if(!a.parameters && !b.parameters) return false;
     return collection_1.StringMapWrapper.equals(a.parameters, b.parameters);
 }
 exports.equalSegments = equalSegments;
